Dataset.py

```diff
@@ -227,6 +228,8 @@ def _load_seq_list_file(filename, use_cache_manager=False, expect_list=True):
 
   def sliding_window(self, xr):
     """
+    TODO: This function is only used inside datasets -> could be protected
```

Comment:
Yes sure.
Why do you add this as a TODO? Why don't you simply do it? (I mean is pretty trivial to check where it is used, and then I think this is a case which is pretty clear. Or if not clear, you can write a short Slack message and ask about it.)